include/accessor.h

```diff
@@ -254,12 +254,24 @@ class accessor<DataT, Dims, Mode, target::device> : public detail::accessor_base
 	}
 
 	template <access_mode M = Mode>
-	inline std::enable_if_t<detail::access::mode_traits::is_producer(M), DataT&> operator[](const id<Dims>& index) const {
-		return m_device_ptr[get_linear_offset(index)];
-	}
-
-	template <access_mode M = Mode>
-	inline std::enable_if_t<detail::access::mode_traits::is_pure_consumer(M), const DataT&> operator[](const id<Dims>& index) const {
+	inline std::conditional_t<detail::access::mode_traits::is_producer(M), DataT&, const DataT&> operator[](const id<Dims>& index) const {
+#if CELERITY_ACCESSOR_BOUNDARY_CHECK
+		// We currently don't support boundary checking for accessors created using accessor_testspy::make_device_accessor,
+		// which does not set m_oob_indices.
+		if(m_oob_indices != nullptr) {
+			const bool is_within_bounds_lo = (index >= m_accessed_virtual_subrange.offset) == detail::id_cast<Dims>(id<3>(true, true, true));
+			const bool is_within_bounds_hi =
+			    (index < (m_accessed_virtual_subrange.offset + m_accessed_virtual_subrange.range)) == detail::id_cast<Dims>(id<3>(true, true, true));
```

Comment:
Maybe move `detail::id_cast<Dims>(id<3>(true, true, true)` to a `constexpr auto all_true` for readability (and same LoC).